### PR TITLE
Run CI on latest OS

### PR DESCRIPTION
**Pull request by fisker**
Run CI on latest OS, and move prod_test to linux, its' faster than macos

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If the change is user-facing) I’ve added my changes to the `CHANGELOG.unreleased.md` file following the template.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
   - job: Dev_Test_Windows
     displayName: Dev Test on Windows Node v12
     pool:
-      vmImage: vs2017-win2016
+      vmImage: "windows-latest"
```
Comment (thorn0) [inline azure-pipelines.yml]: Wasn't it you who wrote recently that pinned versions should be preferred?

**Comment by evilebottnawi** [comment]
@thorn0 i think better use latest os, developers other use latest versions

**Comment by thorn0** [comment]
The test environment should be stable. If it changes constantly, how can we be sure that CI fails are caused by our code, not by something in the OS?

**Comment by evilebottnawi** [comment]
@thorn0 os problems is very rare and we can't fix them on our side, also testing on latest os can help us to catch problems in modules on the latest operating systems

**Comment by thorn0** [comment]
Then we should have tests for both latest and non-latest OS versions to understand whether problems are of that kind.

**Comment by evilebottnawi** [comment]
@thorn0 sounds good, i think we can add latest and non latest jobs to improve testing

**Comment by fisker** [comment]
Setting up github actions https://github.com/prettier/prettier/pull/7431 , so we don't need this anymore.

**Pull request closed without merging**